### ISSUE TITLE
Share guard expressions that differ only by line number nodes.

### ISSUE THREAD

**Issue by gafter** [JuliaServices/Rematch2.jl#63]
Share evaluated results for guard expressions that differ only by line number nodes.

Similarly, we should consider sharing when identical by alpha-renaming.

**Comment by mcmcgrath13** [comment]
Yes, I think these are both good improvements. Lets make sure the README/docs calls out that we do this sharing (cool optimization!) and that because of that when debugging only one call is expected and the variable name might not match what the user anticipated
